packages/react-component-library/src/components/DatePicker/DatePicker.tsx

```diff
@@ -79,49 +63,28 @@ export const DatePicker: React.FC<DatePickerProps> = ({
   isOpen,
 }) => {
   const [state, setState] = useState<StateObject>({
-    startDate,
-    endDate,
-    focusedInput: START_DATE, // START_DATE | END_DATE | null
+    from: startDate,
+    to: endDate,
   })
 
-  function handleDateChange(data: StateObject) {
-    setState({
-      ...data,
-      focusedInput: data.focusedInput || START_DATE,
-    })
+  const { from, to } = state
+  const modifiers = { start: from, end: to }
+
+  function handleDayClick(day: Date) {
+    const newState = isRange
+      ? (DateUtils.addDayToRange(day, state as RangeModifier) as StateObject)
+      : ({ from: day, to: day } as StateObject)
+
+    setState(newState)
 
     if (onChange) {
-      onChange(data)
+      onChange(newState)
```

Comment:
This is a breaking change..?

Reply:
This has been updated so that the old callback signature is now retained.